libyul/CompilabilityChecker.cpp

```diff
@@ -39,7 +39,7 @@ map<YulString, int> CompilabilityChecker::run(
 	bool _optimizeStackAllocation
 )
 {
-	if (EVMDialect const* evmDialect = dynamic_cast<EVMDialect const*>(&_dialect))
+	if (auto const* evmDialect = dynamic_cast<EVMDialect const*>(&_dialect))
```

Comment:
Do we really want to have `auto` even if it is only something like 5 characters shorter? What is the benefit here?
(no need to change anything in this PR, just wanted to start a discussion...)

Reply:
I add these as clang-tidy is producing a warning here, which bothers me personally.
And there were a lot of discussions about this in general. We never ended up with some rule which we follow. Than it is up to person and reviewer every time.

I agree that we should discuss this and decide something whatever it is :D

Reply:
Personally, I would be fine with either version in this case because the type is short but I can see the value in having simpler style rules. If it depends on length or perceived complexity, it gets subjective quickly.

Reply:
I'm not sure how others think, but whenever I read code that has `auto`, my first instinct is trying to figure out what the type behind `auto` is. It is an impulse, I just cannot read the code with the attitude "I do not care what is behind this type". In this case, the type is easy to figure out, but it just is an additional mental step I have to make.

Reply:
I'm the same but for me this is a trade-off with how hard the whole line is to read. If the type is complex and long then using `auto` instead of repeating the type actually helps because the line gets easier to parse as a whole.

Also, being able to enforce the rule with tools is always a good thing and I often see simpler rules favored as long as they are good enough, even at the cost of some readability.